src/main/java/com/iridium/iridiumskyblock/listeners/EntityDamageListener.java

```diff
@@ -3,46 +3,173 @@
+        if (event.getEntity() instanceof Player) {
+            Player victim = (Player) event.getEntity();
+            if (event.getDamager() instanceof Player) {
+                handleDamageBetweenPlayers(event, (Player) event.getDamager(), victim, island.get());
+            } else if (event.getDamager() instanceof Projectile) {
+                Projectile attacker = (Projectile) event.getDamager();
+                if (attacker.getShooter() instanceof Player) {
+                    handleDamageBetweenPlayers(event, (Player) attacker.getShooter(), victim, island.get());
+                } else {
+                    handPlayerDamage(event, island.get());
+                }
+            } else {
+                handPlayerDamage(event, island.get());
+            }
+        } else if (event.getDamager() instanceof Player) {
             Player player = (Player) event.getDamager();
             if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), PermissionType.KILL_MOBS)) {
                 return;
             }
-
             event.setCancelled(true);
-            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
-        }
-
-        if (event.getDamager() instanceof Projectile) {
+            if (hasNoCooldown(player)) {
+                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            }
+        } else if (event.getDamager() instanceof Projectile) {
             Projectile projectile = (Projectile) event.getDamager();
             if (projectile.getShooter() instanceof Player) {
                 Player player = (Player) projectile.getShooter();
                 if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), PermissionType.KILL_MOBS)) {
                     return;
                 }
-
                 event.setCancelled(true);
                 projectile.remove();
-                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                if (hasNoCooldown(player)) {
+                    player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                }
+            }
+        }
+    }
+
+    public void handleDamageBetweenPlayers(EntityDamageByEntityEvent event, Player attacker, Player victim, Island island) {
+        Configuration configuration = IridiumSkyblock.getInstance().getConfiguration();
```

Comment:
If I remember correctly Peaches was against declaring variables for the configs.

Reply:
who matters peaches??